pkg/api/defaults-apiserver.go

```diff
@@ -164,6 +164,13 @@ func (cs *ContainerService) setAPIServerConfig() {
 	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.20.0-alpha.1") && o.KubernetesConfig.APIServerConfig["--service-account-issuer"] == "kubernetes.default.svc" {
 		o.KubernetesConfig.APIServerConfig["--service-account-issuer"] = "https://kubernetes.default.svc.cluster.local"
 	}
+
+	outdatedFeatureGates := []string{}
+	// Remove --feature-gate VolumeSnapshotDataSource starting with 1.22
+	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.22.0-alpha.1") {
+		outdatedFeatureGates = append(outdatedFeatureGates, "VolumeSnapshotDataSource")
```

Comment:
could you please add the link to the PR?
https://github.com/kubernetes/kubernetes/pull/101531

Reply:
here and as part of the PR description